Excel_Adapter/Validation/WorksheetName.cs

```diff
@@ -0,0 +1,197 @@
+                return WorksheetValidation.NotWithinCharacterLimit;
+
+            return issue;
+        }
+
+        /************ Checks *********************/
+
+        private static bool CheckIsUnique(string workSheetName, IXLWorkbook workbook)
+        {
+            return !workbook.Worksheets.Contains(workSheetName);
+        }
+
+        private static bool CheckIsNotReservedWord(string workSheetName)
+        {
+            List<string> reservedWords = new List<string> { "history" };
+
+            return !reservedWords.Contains(workSheetName.ToLower());
+        }
+
+        private static bool CheckIsValidCharacters(string workSheetName)
+        {
+            Regex r = new Regex(@"[\[/\?\]\*\\\:]");
+
+            return !r.IsMatch(workSheetName);
+        }
+
+        private static bool CheckIsNotBeginOrEndWithInvalidCharacter(string workSheetName)
+        {
+            return !workSheetName.StartsWith("\'") && !workSheetName.EndsWith("\'");
+        }
+
+        private static bool CheckIsWithinCharacterLimit(string workSheetName)
+        {
+            return workSheetName.Length <= 31;
+        }
+
+        /************ Modifications *********************/
+        private static string ModifyWorksheetName(string worksheetName, WorksheetValidation issue)
+        {
+            switch (issue)
+            {
+                case WorksheetValidation.ReservedWord:
+                    return SetUniqueNameWithReservedName(worksheetName);
+
+                case WorksheetValidation.Blank:
+                    return SetGeneralName();
+
+                case WorksheetValidation.NotUnique:
+                    return SetUniqueName(worksheetName);
+
+                case WorksheetValidation.BeginOrEndWithInvalidCharacter:
+                    return RemoveInvalidBeginningOrEnding(worksheetName);
+
+                case WorksheetValidation.NotValidCharacters:
+                    return RemoveInvalidCharacters(worksheetName);
+
+                case WorksheetValidation.NotWithinCharacterLimit:
+                    return TrimName(worksheetName);
+
+                default:
+                case WorksheetValidation.Valid:
+                    break;
+            }
+
+            return worksheetName;
+        }
+
+        private static string SetUniqueNameWithReservedName(string worksheetName)
+        {
+            return "BHoM_" + worksheetName;
+        }
+
+        private static string SetGeneralName()
+        {
+            return "BHoM_Export_" + DateTime.Now.ToString("ddMMyy_HHmmss");
+        }
+
+        private static string SetUniqueName(string worksheetName)
+        {
+            return DateTime.Now.ToString("ddMMyy HHmmssf") + "_" + worksheetName;
+        }
+
+        private static string RemoveInvalidBeginningOrEnding(string worksheetName)
+        {
+            if (worksheetName.StartsWith("\'"))
+            {
+                worksheetName = worksheetName.Substring(1, worksheetName.Length - 1);
```

Comment:
```suggestion
                worksheetName = worksheetName.Substring(1);
```

If we want the entire string after the first char then we can just substring like this to avoid the maths on the length.